### ISSUE TITLE
Reading IMAP sent emails

### ISSUE THREAD

**Issue by dvisockas** [mikel/mail#1035]
Is there currently a way to read sent emails?

In _lib/mail/network/retriever_methods/imap.rb_ there is a line `options[:mailbox] ||= 'INBOX'` but changing it to either 'sent' or 'SENT' changes nothing.

**Comment by jeremy** [comment]
What's the name of your "Sent Messages" folder? Is it actually "SENT?"

Try this out with other folders. The Sent folder isn't special; it's a convention for where to keep sent emails and each mail app does it differently. Many allow you to map a folder of your choice for use as Sent, Trash, Junk, etc.

**Comment by dvisockas** [comment]
It's actually SENT.

Then let me rephrase my question - is there a way to get the folder names in a mailbox?

**Comment by jeremy** [comment]
@dvisockas Yep! The Mail lib uses [Ruby's Net::IMAP](http://ruby-doc.org/stdlib-2.3.1/libdoc/net/imap/rdoc/Net/IMAP.html#method-i-list), so you have access to all its methods:
```ruby
# Retrieve the top-level folders by matching with % instead of *
mailboxes = Mail.connection { |imap| imap.list '', '%' }
```
IMAP servers that support [RFC6154](https://tools.ietf.org/html/rfc6154) will label their Sent folder with a `\Sent` flag which we can check for:
```ruby
sent = mailboxes.detect { |m| m.attr.include? :Sent }
```
Other IMAP servers may support the older `XLIST` command which does the same thing:
```ruby
sent = Mail.connection { |imap| imap.xlist '', '%' }.detect { |m| m.attr.include? :Sent }
```
